File: src/Entity/MessagePrive.php
<?php

namespace App\Entity;

use App\Repository\MessagePriveRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: MessagePriveRepository::class)]
#[ORM\Table(name: 'message_prive')]
class MessagePrive
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(type: 'integer')]
    private int $conversationId = 0;

    #[ORM\Column(type: 'integer')]
    private int $senderId = 0;

    #[ORM\Column(type: 'text')]
    private string $contenu = '';

    #[ORM\Column(type: 'time', nullable: true)]
    private ?\DateTimeInterface $sentAt = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getConversationId(): int
    {
        return $this->conversationId;
    }

    public function setConversationId(int $conversationId): static
    {
        $this->conversationId = $conversationId;
        return $this;
    }

    public function getSenderId(): int
    {
        return $this->senderId;
    }

    public function setSenderId(int $senderId): static
    {
        $this->senderId = $senderId;
        return $this;
    }

    public function getContenu(): string
    {
        return $this->contenu;
    }

    public function setContenu(string $contenu): static
    {
        $this->contenu = $contenu;
        return $this;
    }

    public function getSentAt(): ?\DateTimeInterface
    {
        return $this->sentAt;
    }

    public function setSentAt(?\DateTimeInterface $sentAt): static
    {
        $this->sentAt = $sentAt;
        return $this;
    }

}

File: nexora_java/src/main/java/com/pi/entities/MessagePrive.java
package com.pi.entities;

import java.sql.Timestamp;

public class MessagePrive {

    private int id;
    private int conversationId;
    private int senderId;
    private String contenu;
    private Timestamp sentAt;

    public MessagePrive() {}

    public MessagePrive(int conversationId, int senderId, String contenu) {
        this.conversationId = conversationId;
        this.senderId = senderId;
        this.contenu = contenu;
    }

    // getters setters

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }


    public int getConversationId() {
        return conversationId;
    }

    public void setConversationId(int conversationId) {
        this.conversationId = conversationId;
    }

    public int getSenderId() {
        return senderId;
    }

    public void setSenderId(int senderId) {
        this.senderId = senderId;
    }

    public String getContenu() {
        return contenu;
    }

    public void setContenu(String contenu) {
        this.contenu = contenu;
    }

    public Timestamp getSentAt() {
        return sentAt;
    }

    public void setSentAt(Timestamp sentAt) {
        this.sentAt = sentAt;
    }
}
